ansible/roles/vm_set/files/mux_simulator.py

```diff
@@ -382,28 +394,53 @@ def get_mux_bridges(vm_set):
     return mux_bridges
 
 
-@app.route('/mux/<vm_set>', methods=['GET'])
+def get_all_mux_status(vm_set):
+    bridge_prefix = 'mbr-{}-'.format(vm_set)
+    mux_bridges = get_mux_bridges(vm_set)
+    all_mux_status = {}
+    for bridge in mux_bridges:
+        port_index = int(bridge.replace(bridge_prefix, ''))
+        all_mux_status[bridge] = get_mux_status(vm_set, port_index)
+    return all_mux_status
+
+
+def update_all_active_side(all_mux_status, new_active_side):
+    new_all_mux_status = {}
+    for bridge, mux_status in all_mux_status.items():
+        vm_set = mux_status['vm_set']
+        port_index = mux_status['port_index']
+        new_all_mux_status[bridge] = set_active_side(vm_set, port_index, new_active_side)
+    return new_all_mux_status
+
+
+@app.route('/mux/<vm_set>', methods=['GET', 'POST'])
 def all_mux_status(vm_set):
     """Handler for requests to /mux/<vm_set>.
 
     For GET request, return detailed status of all the mux Y cables belong to the specified vm_set.
+    For POST request, update mux active side according to poseted data.
```

Comment:
Could you give a data example for posting data to the whole vm_set? I'm not very clear. Thanks

Reply:
Sure, I'll update.